### PR TITLE
Extract tag if attached to Api level tags

### PR DESCRIPTION
**Pull request by f-alizada**
Added additional case (check if tag is attached to api) when tag should be extracted if apiName setting is provided for the extraction.

### CODE REVIEW

File: tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs [hunk 1]
```diff
@@ -129,5 +129,74 @@ public async Task GenerateTagTemplates_GetApiOperationRelated_ProperlyLaysTheInf
...
+                this.GetTestLogger<TagExtractor>(),
+                mockedTagClient,
+                new TemplateBuilder());
+
+            var extractorExecutor = ExtractorExecutor.BuildExtractorExecutor(
+                this.GetTestLogger<ExtractorExecutor>(),
+                tagExtractor: tagExtractor);
+            extractorExecutor.SetExtractorParameters(extractorParameters);
+
+            ApiTemplateResources apiTemplateResources = new ApiTemplateResources()
```
Comment (DeagleGross) [inline tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs:153]: nit: var?

File: tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs [hunk 2]
```diff
@@ -129,5 +129,74 @@ public async Task GenerateTagTemplates_GetApiOperationRelated_ProperlyLaysTheInf
...
+            {
+                Tags = new List<TagTemplateResource>()
```
Comment (DeagleGross) [inline tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs:155]: nit: remove curly braces if you still use {} for object initialization?

Comment (DeagleGross) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs:155]: and lower also

File: tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs (hunk 1 above)
Comment (DeagleGross) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs:153]: and lower also

**Review by DeagleGross — DISMISSED** [review]

File: tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs (hunk 1 above)
Comment (f-alizada) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs]: Thank you, updated

File: tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs (hunk 2 above)
Comment (f-alizada) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/TagExtractorTests.cs]: thank you! Agree updated!

**Review by DeagleGross — APPROVED** [review]

**Pull request merged**